### PR TITLE
Fix XmlWriter Outerloop test failures

### PR DESCRIPTION
**Pull request by sepidehMS**
Fixes https://github.com/dotnet/corefx/issues/6331 and https://github.com/dotnet/corefx/issues/1491

These tests were failing due to: **1)** Newline differences on Unix and Windows, **2)** Decimal point differences in some cultures

cc: @danmosemsft @stephentoub @AlexGhiondea

### CODE REVIEW

**Comment by sepidehMS** [comment]
@dotnet-bot test Outerloop Windows_NT Debug please
@dotnet-bot test Outerloop Ubuntu14.04 Debug please
@dotnet-bot test Outerloop OSX Debug please

**Comment by sepidehMS** [comment]
@dotnet-bot test Outerloop Windows_NT Debug please
@dotnet-bot test Outerloop Ubuntu14.04 Debug please
@dotnet-bot test Outerloop OSX Debug please

**Comment by sepidehMS** [comment]
@dotnet-bot test Outerloop Windows_NT Debug please
@dotnet-bot test Outerloop Ubuntu14.04 Debug please
@dotnet-bot test Outerloop OSX Debug please

**Comment by sepidehMS** [comment]
@dotnet-bot test Outerloop Windows_NT Debug please
@dotnet-bot test Outerloop Ubuntu14.04 Debug please
@dotnet-bot test Outerloop OSX Debug please

**Comment by sepidehMS** [comment]
@dotnet-bot test Outerloop OSX Debug please

**Comment by sepidehMS** [comment]
@dotnet-bot test Outerloop Ubuntu14.04 Debug please

**Comment by sepidehMS** [comment]
@dotnet-bot test Outerloop Windows_NT Debug please
@dotnet-bot test Outerloop Ubuntu14.04 Debug please
@dotnet-bot test Outerloop OSX Debug please

**Comment by sepidehMS** [comment]
@dotnet-bot test Outerloop Windows_NT Debug please
@dotnet-bot test Outerloop Ubuntu14.04 Debug please
@dotnet-bot test Outerloop OSX Debug pleas

**Comment by sepidehMS** [comment]
@danmosemsft I'm not adding anything but comments in this PR right now, but the OuterLoop OSX Debug run still fails. Is there anybody who can investigate this?

**Comment by danmosemsft** [comment]
@mmitche

**Comment by mmitche** [comment]
@danmosemsft @sepidehMS These look like real failures:

https://ci.dot.net/job/dotnet_corefx/job/master/job/outerloop_osx_debug_prtest/9/consoleFull#84395483279fe3b83-f408-404c-b9e7-9207d232e5fc

**Comment by mmitche** [comment]
@sepidehMS @danmosemsft Specifically, the few lines above that show:

/Users/dotnet-bot/j/workspace/dotnet_corefx/master/outerloop_osx_debug_prtest/bin/tests/Unix.AnyCPU.Debug/Common.Tests/default.netcoreapp1.1/RunTests.sh: line 209: ./corerun: No such file or directory

**Comment by mmitche** [comment]
Also looking at the non-PR version, I see you have the same failures in the checked in sources:

https://ci.dot.net/job/dotnet_corefx/job/master/job/outerloop_osx_debug/

This is the first failing build, and lists the associated changes

https://ci.dot.net/job/dotnet_corefx/job/master/job/outerloop_osx_debug/30/

**Comment by sepidehMS** [comment]
@karajas Looking at https://ci.dot.net/job/dotnet_corefx/job/master/job/outerloop_osx_debug/30/, do you have any idea what can be causing this? I cannot repro this to see what's wrong.

**Comment by karajas** [comment]
@sepidehMS I _think_ it may be because the wrong TestNugetRuntimeId is being passed. Looks like in Innerloop tests we pass in `[17:44:25.05] Using RuntimeIdentifier = 'osx.10.10-x64' ` during the build-tests phase and had to add some additional logic to build.sh to ensure this.

From what I can tell it looks like Outerloop OSX is using ` [23:21:21.05] Using RuntimeIdentifier = 'osx.10.11-x64' ` which tries to restore for the wrong rid. I would try passing in /p:TestNugetRuntimeId=osx.10.10-x64 to the outerloop osx build. /cc @weshaggard

**Comment by karajas** [comment]
@sepidehMS you can add TestNugetRuntimeId here https://github.com/dotnet/corefx/blob/master/netci.groovy#L235.

Looks like the ubuntu outerloop is already doing it. https://github.com/dotnet/corefx/blob/master/netci.groovy#L238

**Comment by sepidehMS** [comment]
Thanks a lot @karajas! I'm going to change that in a separate PR.

**Comment by sepidehMS** [comment]
@dotnet-bot test Outerloop Windows_NT Debug please
@dotnet-bot test Outerloop Ubuntu14.04 Debug please
@dotnet-bot test Outerloop OSX Debug pleas

**Comment by sepidehMS** [comment]
@dotnet-bot test OuterLoop Ubuntu14.04 Debug please

**Comment by sepidehMS** [comment]
PTAL

File: src/System.Private.Xml/tests/Writers/XmlWriterApi/TCFullEndElement.cs [hunk 1]
```diff
@@ -640,8 +640,8 @@ public override void AddChildren()
                 // for function CData_13
                 {
                     this.AddChild(new CVariation(CData_13) { Attribute = new Variation("WriteCData with 0x0A with NewLineHandling.Entitize") { Params = new object[] { 10, NewLineHandling.Entitize, "<r><![CDATA[\n]]></r>" }, id = 18, Pri = 1 } });
-                    this.AddChild(new CVariation(CData_13) { Attribute = new Variation("WriteCData with 0x0D with NewLineHandling.Replace") { Params = new object[] { 13, NewLineHandling.Replace, "<r><![CDATA[\r\n]]></r>" }, id = 13, Pri = 1 } });
```
Comment (danmosemsft) [inline src/System.Private.Xml/tests/Writers/XmlWriterApi/TCFullEndElement.cs]: I assume you've searched for \r\n everywhere in the XML tests

Comment (sepidehMS) [inline src/System.Private.Xml/tests/Writers/XmlWriterApi/TCFullEndElement.cs]: Here I actually only fixed the failing outerloop tests. I'm working on cleaning up all Xml tests (https://github.com/dotnet/corefx/issues/14071) right now and have that change in there.

**Comment by sepidehMS** [comment]
@dotnet-bot test Outerloop Windows_NT Debug please
@dotnet-bot test Outerloop Ubuntu14.04 Debug please
@dotnet-bot test Outerloop OSX Debug please

**Comment by AlexGhiondea** [comment]
LGTM.

File: src/System.Private.Xml/tests/Writers/XmlWriterApi/CommonTests.cs [hunk 2]
```diff
@@ -7108,6 +7108,7 @@ public int writeValue_27()
                 Type dest = typeMapper[destStr];
                 bool isValid = (bool)CurVariation.Params[3];
                 object expVal = (object)CurVariation.Params[4];
+                CultureInfo origCulture = null;
```
Comment (stephentoub) [inline src/System.Private.Xml/tests/Writers/XmlWriterApi/CommonTests.cs]: Nit: why not define and set this just before the try block, e.g.
```C#
CultureInfo origCulture = CultureInfo.CurrentCulture;
try
{
    CultureInfo.CurrentCulture = CultureInfo.InvariantCulture;
    VerifyValue(dest, expVal, param);
}
finally
{
    CultureInfo.CurrentCulture = origCulture;
}
```
?

**Pull request merged**